xapian-applications/omega/query.cc

```diff
@@ -752,11 +765,12 @@ html_highlight(const string &s, const string &list,
 	}
 	j = term_end;
 	term = Xapian::Unicode::tolower(term);
-	int match = word_in_list(term, list);
+	WordList w;
+	int match = w.word_in_list(term);
```

Comment:
This confusingly depends on you having previously created a `WordList` and called `build_word_map` on it, or else the mapping would be empty.

It would be more obvious how this worked to if you either pass in `list` as before and construct the `WordList` in this function, or pass in the `WordList` object that the caller constructs, or have a single static `WordList` object rather than constructing two for every `$highlight` expanded.